routing/pathfind.go

```diff
@@ -260,6 +274,47 @@ func newRoute(sourceVertex route.Vertex,
 		nextIncomingAmount = amtToForward + fee
 	}
 
+	// If we are creating a route to a blinded path, we need to add some
+	// additional data to the route that is required for blinded forwarding.
+	// We do another pass on our edges to append this data.
+	if blindedPath != nil {
+		var (
+			inBlindedRoute bool
+			dataIndex      = 0
+
+			introVertex = route.NewVertex(
+				blindedPath.IntroductionPoint,
+			)
+		)
+
+		for i, hop := range hops {
+			// Once we locate our introduction node, we know that
+			// every hop after this is part of the blinded route.
+			if bytes.Equal(hop.PubKeyBytes[:], introVertex[:]) {
```

Comment:
Since we're ranging through hops forward, we'll always hit the `IntroductionPoint` first here (I think it's fair to trust our pathfinding to not include one node more than once).

> guess because e(i+1) = H(E(i) || ss(i)) * e(i), e(i+1)

But yeah, well formed blinded routes will be unique!